### ISSUE TITLE
Search filter is case sensitive for non ASCII characters

### ISSUE THREAD

**Issue by JackGruber** [laurent22/joplin#5102]
When you have tags/notebooks with mixed case and a non ASCII char is uppercase/lowercase you can't search for them with the wrong case in the search filter.

Affected filter:
- tag
- notebook
 
## Environment

Joplin 2.1.0 (dev, win32)

Sync Version: 2
Profile Version: 39
Keychain Supported: Yes

### Tag
1. Import Notes with mixedcase or upper case tags like `AÜW`
2. Search: `tag:aüw` = no search results
3. Search: `tag:aÜw` = search results

### Notebook
1. Create a notebook `AÜW`
2. Create some notes in the notebook
2. Search: `notebook:aüw` = no search results
3. Search: `notebook:aÜw` = search results

## Describe what you expected to happen

You can filter even if you don't know the exact casing.

## More info

(18) Case-insensitive matching of Unicode characters does not work.
https://www.sqlite.org/faq.html#q18

**SQL Example**: 
`SELECT LOWER('MY TEST STRING with non ASCII ÜÄÖÉ ...') AS test; `

| test |
| --- |
| my test string with non ascii char ÜÄÖÉ ... |

One possible solution would be to overload the default LOWER() function.

**Comment by laurent22** [comment]
In the normalizeText function of the search engine, I see the diacritics are already removed so in theory it should work. But maybe for some reason certain letters aren't processed properly.

**Comment by JackGruber** [comment]
Tags and notebook filter directly on the `tags` and `folder` table with `LIKE`. 
If I see it correctly, only the note text is normalised.

**Comment by laurent22** [comment]
Good point. That might be trickier than I thought then as if we can't use the LOWER function, it means we need to normalize folder and tag titles in a separate table too.
